feelslikemonday/app/src/main/java/com/example/feelslikemonday/ui/followrequest/ArraryAdapter/RequestList.java

```diff
@@ -1,43 +1,139 @@
+import com.example.feelslikemonday.DAO.FollowRequestCallback;
+import com.example.feelslikemonday.DAO.FollowRequestDAO;
+import com.example.feelslikemonday.DAO.VoidCallback;
 import com.example.feelslikemonday.R;
+import com.example.feelslikemonday.model.FollowPermission;
+import com.example.feelslikemonday.model.FollowRequest;
 import com.example.feelslikemonday.model.User;
 
 import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
 
-public class RequestList extends ArrayAdapter<User> {
-    private ArrayList<User> users;
+public class RequestList extends ArrayAdapter<String> {
     private Context context;
+    private List<String> requesterUsernames;
+    private List<String> followeeUsernames;
+    private static FollowPermissionDAO DAO;
+    private static FollowRequestDAO followRequestDAO;
+    private final CountDownLatch signal = new CountDownLatch(1);
 
 
-    public RequestList(@NonNull Context context, ArrayList<User> users) {
+    public RequestList(@NonNull Context context, ArrayList<String> users) {
         super(context, 0, users);
-        this.users = users;
+        this.requesterUsernames = users;
         this.context = context;
     }
 
     @NonNull
     @Override
     public View getView(int position, @Nullable View convertView, @NonNull ViewGroup parent) {
         //return super.getView(position, convertView, parent);
+        DAO = FollowPermissionDAO.getInstance();
+        followRequestDAO = FollowRequestDAO.getInstance();
         View view = convertView;
+
         if(view == null){
             view = LayoutInflater.from(context).inflate(R.layout.content_follow_request, parent, false);
         }
-        User user = users.get(position);
+        final String user = requesterUsernames.get(position);
+
         TextView userName = view.findViewById(R.id.username_text);
-        userName.setText(user.getUsername());
+        userName.setText(user);
+
+        Button acceptButton = view.findViewById(R.id.request_accept_button);
+        Button rejectButton = view.findViewById(R.id.request_reject_button);
+
+        acceptButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                Log.i("testListView", "clickAcceptButton");
+
+                DAO.get("xiaole", new FollowPermissionCallback(){
+                    @Override
+                    public void onCallback(FollowPermission followPermission) {
+                        if(followPermission.getFollowerUsername().equals("xiaole")){
+                            followeeUsernames = followPermission.getFolloweeUsernames();
+                            followeeUsernames.add(user);
+
+                            DAO.createOrUpdate("xiaole", followPermission, new VoidCallback() {
+                                @Override
+                                public void onCallback() {
+                                    signal.countDown();
```

Comment:
I dont think this signal.countDown() method should be used in non-test code. Christian wrote something about this in the tests - could you double check with Christian about this?